src/main/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/PaymentTaskHandler.java

```diff
@@ -4,41 +4,56 @@
 import org.camunda.bpm.client.task.ExternalTask;
-import org.camunda.bpm.client.task.ExternalTaskHandler;
-import org.camunda.bpm.client.task.ExternalTaskService;
+import org.camunda.bpm.engine.variable.VariableMap;
+import org.camunda.bpm.engine.variable.Variables;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
 import uk.gov.hmcts.reform.ccd.client.model.Event;
 import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
-import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
 import uk.gov.hmcts.reform.unspec.helpers.CaseDetailsConverter;
 import uk.gov.hmcts.reform.unspec.model.BusinessProcess;
 import uk.gov.hmcts.reform.unspec.model.CaseData;
 import uk.gov.hmcts.reform.unspec.service.CoreCaseDataService;
 import uk.gov.hmcts.reform.unspec.service.data.ExternalTaskInput;
+import uk.gov.hmcts.reform.unspec.service.flowstate.StateFlowEngine;
 
 import java.util.Map;
 
 @Component
 @Slf4j
 @RequiredArgsConstructor
-public class PaymentTaskHandler implements ExternalTaskHandler {
+public class PaymentTaskHandler implements BaseExternalTaskHandler {
 
+    public static final String FLOW_STATE = "flowState";
     private final CoreCaseDataService coreCaseDataService;
     private final CaseDetailsConverter caseDetailsConverter;
     private final ObjectMapper objectMapper;
+    private final StateFlowEngine stateFlowEngine;
+
+    private CaseData data;
 
     @Override
-    public void execute(ExternalTask externalTask, ExternalTaskService externalTaskService) {
+    public void handleTask(ExternalTask externalTask) {
         Map<String, Object> allVariables = externalTask.getAllVariables();
         ExternalTaskInput externalTaskInput = objectMapper.convertValue(allVariables, ExternalTaskInput.class);
         String caseId = externalTaskInput.getCaseId();
-        CaseEvent caseEvent = externalTaskInput.getCaseEvent();
-        StartEventResponse startEventResponse = coreCaseDataService.startUpdate(caseId, caseEvent);
-        CaseData data = caseDetailsConverter.toCaseData(startEventResponse.getCaseDetails());
-        BusinessProcess businessProcess = data.getBusinessProcess().updateActivityId(externalTask.getActivityId());
-        coreCaseDataService.submitUpdate(caseId, caseDataContent(startEventResponse, businessProcess));
-        externalTaskService.complete(externalTask);
+        StartEventResponse startEventResponse = coreCaseDataService.startUpdate(caseId,
+                                                                                externalTaskInput.getCaseEvent());
+        BusinessProcess businessProcess = caseDetailsConverter.toCaseData(startEventResponse.getCaseDetails())
+            .getBusinessProcess().updateActivityId(externalTask.getActivityId());
+        data = coreCaseDataService.submitUpdate(caseId, caseDataContent(startEventResponse, businessProcess));
+    }
+
+    @Override
+    public VariableMap getVariableMap() {
```

Comment:
Could we move this up in the parent class - I think all external task would benefit from having the state flow in context?

Reply:
We've got one parent class for all tasks, including CaseStayed and PollingEventEmitter where above logic would be invalid.